partiql-planner/src/main/kotlin/org/partiql/planner/internal/problems/AlwaysMissing.kt

```diff
@@ -0,0 +1,20 @@
+package org.partiql.planner.internal.problems
+
+import org.partiql.spi.SourceLocation
+import org.partiql.spi.errors.Classification
+import org.partiql.spi.errors.PError
+import org.partiql.spi.errors.Severity
+
+/**
+ * @param location see [PError.location]
+ * @return an error representing [PError.ALWAYS_MISSING]
+ */
+internal class AlwaysMissing(
+    location: SourceLocation?
+) : PError(
+    ALWAYS_MISSING,
+    Severity.WARNING(),
+    Classification.SEMANTIC(),
+    location,
+    null
+)
```

Comment:
Rather than a bunch of extra classes, how about simplifying to just static methods? We already have codes so the additionally classes don't help us and they do not extend functionality of the PError class. The constructors here are effectively being used as named factory methods, so let's do just that. This would allow PError to be final as well.

_Errors.kt_
```kotlin

internal object Errors {

    @JvmStatic
    fun alwaysMissing(location: SourceLocation) = PError(
       ALWAYS_MISSING,
       Severity.WARNING(),
       Classification.SEMANTIC(),
       location,
       null,
    )

   // a place to instantiate errors
}

```